src/daemon/daemonlocalserverconnection.cpp

```diff
@@ -160,3 +171,60 @@ void DaemonLocalServerConnection::write(const QJsonObject& obj) {
   m_socket->write(QJsonDocument(obj).toJson(QJsonDocument::Compact));
   m_socket->write("\n");
 }
+
+bool DaemonLocalServerConnection::isCallerAuthorized(const QString& command) {
+#if defined(MZ_MACOS)
+  uid_t uid;
+  gid_t gid;
+  int result = getpeereid(m_socket->socketDescriptor(), &uid, &gid);
+  if (result == -1) {
+    logger.error() << "Unable to determine if caller is authorized."
+                      "Assuming unauthorized.";
+    return false;
+  }
+
+  if (uid == 0) {
+    logger.debug() << "Command issued by root.";
+    // Case: This is root calling, root is always authorized.
+    return true;
+  }
+
+  QJsonObject status = Daemon::instance()->getStatus();
+  bool isConnected = status.value("connected").toBool();
+
+  if (isConnected) {
+    if (m_sessionUid == uid) {
+      // Case: VPN is currently active and caller is the one who activated it.
+      // Caller can access the APIs.
+      return true;
+    }
+
+    // Case: VPN is currently active and caller is **not** one who activated it.
+    // Caller **can't** access the APIs.
+    logger.error() << "Command issued by unexpected user" << uid;
+    return false;
+  }
+
+  // Case: VPN is not connected and we got a request to check the status, fine.
+  if (command == "status") {
```

Comment:
The no connection check returns an object with the only preoperty `connection`:`false`, while the connected check returns an object with a bit more information about the connection (see: https://github.com/mozilla-mobile/mozilla-vpn-client/blob/main/src/daemon/daemon.cpp#L445-L474). I am not really sure if that is a concern, however I am going with the principle of least priviledge possible here just in case.